engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -911,7 +911,7 @@ public function getEntityTypeSubtypeWhereSql($table, $types, $subtypes, $pairs)
 				$types = array($types);
 			}
 
-			if ($subtypes && !is_array($subtypes)) {
+			if (!is_array($subtypes)) {
```

Comment:
There is no typehint so it could be false or null, not sure how normalisation works then. Maybe add a test to be sure

Reply:
`null`, `''`, `0` are handled a bit later on https://github.com/Elgg/Elgg/blob/2.3/engine/classes/Elgg/Database/EntityTable.php#L950-L957